.circleci/test.yml

```diff
@@ -0,0 +1,313 @@
+            sudo apt-get install -y ruby2.7
+            pip install pre-commit
+            pre-commit install
+      - run:
+          name: Linting
+          command: pre-commit run --all-files
+      - run:
+          name: Check docstring coverage
+          command: |
+            pip install interrogate
+            interrogate -v --ignore-init-method --ignore-module --ignore-nested-functions --ignore-regex "__repr__" --fail-under 80 mmdeploy
+  setup_environment:
+    parameters:
+      executor_name:
+        type: string
+      python:
+        type: string
+        default: 3.7.4
+      torch:
+        type: string
+      torchvision:
+        type: string
+      mmcv:
+        type: string
+        default: 1.4.0
+      mmcv_url_index:
+        type: string
+    executor: << parameters.executor_name >>
+    steps:
+      - attach_workspace:
+          at: mmdeploy
+      - restore_cache:
+          key: v3-<< parameters.executor_name >>-<< parameters.python >>-<< parameters.torch >>-<<parameters.mmcv >>
+      - run:
+          name: Install Python, Pytorch and mmcv-full
+          command: |
+            bash mmdeploy/.circleci/scripts/install_python.sh << parameters.python >>
+            conda/bin/pip install torch==<< parameters.torch >> torchvision==<< parameters.torchvision >> -f https://download.pytorch.org/whl/torch_stable.html
+            conda/bin/pip install mmcv-full==<< parameters.mmcv >> -f https://download.openmmlab.com/mmcv/dist/<< parameters.mmcv_url_index >>/index.html
+      - save_cache:
+          paths:
+            - conda
+          key: v3-<< parameters.executor_name >>-<< parameters.python >>-<< parameters.torch >>-<<parameters.mmcv>>
+  checkout_mmdeploy:
+    parameters:
+      executor_name:
+        type: string
+    executor: << parameters.executor_name >>
+    steps:
+      - checkout_full
+      - persist_to_workspace:
+          root: mmdeploy
+          paths:
+            - .
+  build_mmdeploy_cpu:
+    parameters:
+      executor_name:
+        type: string
+      python:
+        type: string
+        default: 3.7.4
+      torch:
+        type: string
+      mmcv:
+        type: string
+        default: 1.4.0
+    executor: << parameters.executor_name >>
+    steps:
+      - attach_workspace:
+          at: mmdeploy
+      - restore_cache:
+          key: v3-<< parameters.executor_name >>-<< parameters.python >>-<< parameters.torch >>-<<parameters.mmcv>>
+      - run:
+          name: Install SDK dependencies
+          command: |
+            sudo apt-get update
+            sudo apt-get install libopencv-dev libspdlog-dev
+      - run:
+          name: Install ONNXRuntime
+          command: |
+            bash mmdeploy/.circleci/scripts/install_onnxruntime.sh
+            export ONNXRUNTIME_DIR=$(pwd)/onnxruntime
+            echo "export ONNXRUNTIME_DIR=${ONNXRUNTIME_DIR}" >> ~/.bashrc
+            echo "export LD_LIBRARY_PATH=$ONNXRUNTIME_DIR/lib:$LD_LIBRARY_PATH" >> ~/.bashrc
```

Comment:
I see environment variables are being exported many times. It could be optimized a bit I guess.

https://github.com/open-mmlab/mmocr/blob/main/.circleci/test.yml#L50-L51

On CircleCI, you can export everything to `$BASH_ENV` and `source $BASH_ENV` to update the environment variables that the following commands will use. Don't forget the dot before the script path to ensure the script will be running on the current bash.

Reply:
I tried "source $BASH_ENV" in windows VM, but got failure like,
```shell
source $BASH_ENV
source : The term 'source' is not recognized as the name of a cmdlet, function, script file, or operable program. 
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ source $BASH_ENV
+ ~~~~~~
    + CategoryInfo          : ObjectNotFound: (source:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```